auto-restart (#1648)
kubernetes-sigs/secrets-store-csi-driver: docs/book/src/topics/auto-restart-on-rotation.md

# Auto restart of pods when secret is rotated

_This is a suggested implementation which can be used alongside the Secrets Store CSI Driver._

When [auto rotation of secrets](./secret-auto-rotation.md) is enabled, workloads which depend upon secrets will need to either 
- watch for updates to secrets and reload these in their runtime, or
- be restarted to pick up the latest secrets when they change

> A solution such as [Reloader](https://github.com/stakater/Reloader) could be used to watch for updates to Kubernetes Secrets or ConfigMaps and restart pods when a change is detected. However, if secret values are mounted as volumes in the pods, that solution is not suitable.

Reviewer: From an official SIG-Auth project, should we be linking out to this in the formal docs?

Reviewer: This seems in tone a bit like a blog post for solving a particular problem, rather than a docs site entry. 🤔

Reviewer: Per our discussion in the sync meeting, perhaps we keep, but just note that this is not supported or tested.

Author: You could see this doc as a continuation from https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/main/docs/book/src/topics/secret-auto-rotation.md.

I aimed for similar tone.

That doc links to Reloader.

I'm open to suggestions.

My motivation here is that the solution I arrived at and implemented was
- not obvious based on existing resources 
- fairly straightforward given the necessary pointers

Author: @benjaminapetersen I have added a note to the top of the doc. If this is not what you have in mind, could you suggest alternative? Thanks

> A solution such as [Reloader](https://github.com/stakater/Reloader) could be used to watch for updates to Kubernetes Secrets or ConfigMaps and restart pods when a change is detected. However, if secret values are mounted as volumes in the pods, that solution is not suitable.

Reviewer 2: ```suggestion
A solution like [Reloader](https://github.com/stakater/Reloader) can watch updates to Kubernetes Secrets or ConfigMaps and restart pods when a change is detected. However, if secret values are mounted as volumes in the pods, that solution is not suitable.
```

> Using custom resources created by the Secrets Store CSI Driver, a Kubernetes Controller can be used to detect when secrets are updated by the driver and to restart affected pods.

Reviewer 2: ```suggestion
Using custom resources created by the Secrets Store CSI Driver, a Kubernetes controller can detect when secrets are updated by the driver and restart the associated pods.
```

## SecretProviderClassPodStatus custom resource

> The relevant custom resource is [`SecretProviderClassPodStatus`](https://secrets-store-csi-driver.sigs.k8s.io/concepts#secretproviderclasspodstatus).

Reviewer 2: ```suggestion
The relevant custom resource is [`SecretProviderClassPodStatus`](./concepts.md#secretproviderclasspodstatus).
```

something like this, I haven't tested it, so please confirm it works.

Author: committed working link

> Each `SecretProviderClassPodStatus` CR has a one-to-one relationship with a pod which references secrets using the Secrets Store CSI Driver. The CR identifies the pod name, namespace and other attributes. The driver manages the lifecyle of `SecretProviderClassPodStatus` which is linked to the lifecycle of the affected pod.

Reviewer 2: ```suggestion
Each `SecretProviderClassPodStatus` custom resource (CR) has a one-to-one relationship with a pod that references secrets using the Secrets Store CSI Driver. The CR includes the pod name, namespace and other attributes. The driver manages the lifecycle of `SecretProviderClassPodStatus` which is tied to the lifecycle of the associated pod.
```

> Each `SecretProviderClassPodStatus` CR has a one-to-one relationship with a pod which references secrets using the Secrets Store CSI Driver. The CR identifies the pod name, namespace and other attributes. The driver manages the lifecyle of `SecretProviderClassPodStatus` which is linked to the lifecycle of the affected pod.

Reviewer 2: It's 1:1 mapping for pod and SPC, so clarify that here

Author: You have, or I should?

Author: Done
